ipv8/messaging/anonymization/tunnelcrypto.py

```diff
@@ -43,12 +43,12 @@ def generate_diffie_shared_secret(self, dh_received, key=None):
         Y = tmp_key.key.pk
         shared_secret = libnacl.crypto_box_beforenm(dh_received, y) + libnacl.crypto_box_beforenm(dh_received, key.key.sk)
 
-        AUTH = libnacl.crypto_auth(Y, shared_secret)
+        AUTH = libnacl.crypto_auth(Y, shared_secret[:32])
```

Comment:
This seems equivalent to not appending `libnacl.crypto_box_beforenm(dh_received, key.key.sk)` to the shared secret. Was the remainder of the message never checked by `libnacl` in the old version?

Seems kind of fishy, please provide 4 unit tests with `1.6` -> `1.6`, `1.6` -> `1.7`, `1.7` -> `1.6` and `1.7` -> `1.7` behavior.

Reply:
The `shared_secret` is also used for generating session keys, not just for the MAC of the `created` cell.

In `1.6`, the function apparently just drops any data beyond the first 32 bytes. I would prefer calculating the MAC using a different key, but that would break compatibility. Since I'm already planning to break to the protocol for Tribler 7.5, we could just wait until then.

Reply:
Alright, if you checked it, that's fine. Please make a sister PR to the `asyncio` branch so we can merge this.